src/main.zig

```diff
@@ -1420,30 +1430,40 @@ fn buildOutputType(
                     // We don't need to care about these because these args are
                     // for resolving circular dependencies but our linker takes
                     // care of this without explicit args.
-                } else if (mem.startsWith(u8, arg, "--major-os-version") or
-                    mem.startsWith(u8, arg, "--minor-os-version"))
+                } else if (mem.eql(u8, arg, "--major-os-version") or
+                    mem.eql(u8, arg, "--minor-os-version"))
```

Comment:
This breaks `--major-os-version=foo`

Reply:
I don't understand. Previous to PR these options were no-op but `startsWith` creates some problems:
- accepts `--major-os-version`, `--major-os-version=`, and `--major-os-versionANYTHING`
- if `--major-os-version` is used it is likely the user read the man page and used 2-arg syntax, and in that case following arg is not consumed

afaict this args don't do anything anyways and IMO we should accept them in both 2-arg and 1-arg-with-equals syntaxes to properly eat the command line args.